src/scripts/domParsers/khanAcademyVideoExtractor.ts

```diff
@@ -0,0 +1,50 @@
+import {VideoExtractor} from "./videoExtractor";
+import {VideoUtils} from "./videoUtils";
+import {YoutubeVideoExtractor} from "./youtubeVideoExtractor";
+
+import {Utils} from "../utils";
+
+export class KhanAcademyVideoExtractor implements VideoExtractor {
+	private youtubeExtractor: YoutubeVideoExtractor;
```

Comment:
Something about this `KhanAcademyVideoExtractor` actually holding a `YoutubeVideoExtractor` makes me wonder: how about `KhanAcademyVideoExtractor extends YoutubeVideoExtractor`? 

I know it may feel kind of pointless because we'd override most of the `YoutubeVideoExtractor` method implementations... but hey, then `KhanAcademyVideoExtractor.createEmbeddedVideos` can call `super.createEmbeddedVideos` instead :D

More seriously, I also feel like using this pattern would make it quicker to grasp what's going on here